contrib_bots/lib/poll.py

```diff
@@ -0,0 +1,105 @@
+
+
+class PollHandler(object):
+    '''
+    This plugin facilitates polling.
+    '''
+
+    def __init__(self):
+        # polls is a dict of dict. First dict corresponds to poll title,
+        # second refers to answers in poll
+        self.polls = dict()
+
+        # streams is a convenience object that keeps track of the latest
+        # question for each stream.
+        self.streams = dict()
```

Comment:
Maybe instead we could have one poll at the time, that begins with `@poll init` and runs until `@poll close` - we wouldn't have to use `poll title` to identify a specific poll - other than retrieving results for older polls.

We could manage older polls by having a `@poll archive` option (name up for discussion) and the user could then use the `<question>` to get old polls. Or simply a `@poll list` with all the old questions and their results.

What do you think? I think we need to close the polls anyway, so the `@poll close` is definitely something to add.